src/gui/resourcetree.h

```diff
@@ -0,0 +1,143 @@
+/* Phaethon - A FLOSS resource explorer for BioWare's Aurora engine games
+ *
+ * Phaethon is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * Phaethon is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * Phaethon is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Phaethon. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Phaethon's tree of game resource files.
+ */
+
+#ifndef RESOURCETREE_H
+#define RESOURCETREE_H
+
+#include <QAbstractItemModel>
+#include <QFileIconProvider>
+#include <QFileInfo>
+#include <QModelIndex>
+#include <QVariant>
+
+#include "verdigris/wobjectdefs.h"
+
+#include "src/aurora/archive.h"
+#include "src/aurora/util.h"
+
+#include "src/common/filetree.h"
+#include "src/common/ptrmap.h"
+
+#include "src/images/decoder.h"
+
+namespace Common {
+    class SeekableReadStream;
+}
+
+namespace Sound {
+	class AudioStream;
+}
+
+namespace Images {
+	class Decoder;
+}
+
+namespace Aurora {
+	class KEYFile;
+	class KEYDataFile;
+}
+
+namespace GUI {
+
+struct Archive;
+class MainWindow;
+class ResourceTreeItem;
+
+class ResourceTree : public QAbstractItemModel {
+	W_OBJECT(ResourceTree)
+
+public:
+	ResourceTree(MainWindow *mainWindow, QObject *parent = 0);
+	~ResourceTree();
+
+	void populate(const Common::FileTree::Entry &rootEntry);
+	void populate(const Common::FileTree::Entry &rootEntry, ResourceTreeItem *parent);
+
+	/** Called in fetchMore.
+	*/
+	void insertItemsFromArchive(Archive &archive, const QModelIndex &parentIndex);
+	void insertItems(size_t position, QList<ResourceTreeItem *> &items, const QModelIndex &parentIndex);
+
+	Aurora::Archive     *getArchive(const QString &path);
+	Aurora::KEYDataFile *getKEYDataFile(const QString &file);
+	void                loadKEYDataFiles(Aurora::KEYFile &key);
+
+	/** Return the item in the tree structure that corresponds to the given index.
+	*/
+	ResourceTreeItem *itemFromIndex(const QModelIndex &index) const;
+
+	/** Model functions. */
+
+	/** Return the index if it exists, else create it.
+	*/
```

Comment:
Please remove these line breaks inside the Doxygen comments here. There's no need to make the comment span two lines.

This applies to resourcetreeitem.h as well and further commits as well.